web/src/games/kriegspiel/Board.tsx

```diff
@@ -0,0 +1,770 @@
+    if (id === pickedID) {
+      return pico8Palette.dark_purple;
+    } else if (isAvailable(id)) {
+      //predict supply after moving
+      if (
+        getSuppliedCells(
+          {
+            ...G,
+            cells: G.cells.map((obj, CId) => (CId === pickedID ? null : CId === id ? pickedData(pickedID) : obj)),
+          },
+          currentPlayer,
+        ).includes(id)
+      ) {
+        return pico8Palette.green;
+      } else {
+        return pico8Palette.yellow;
+      }
+    } else if (typeof strongholdColor === 'string') {
+      return fictionColor(strongholdColor);
+    } else {
+      const pos = CId2Pos(id);
+      const colorCase = (pos.x + pos.y) % 2 === 0;
+      return colorCase ? pico8Palette.light_grey : pico8Palette.white;
+    }
+  }
+
+  function renderBattleEffect(CId: CellID, selected: boolean) {
+    const obj = G.cells[CId];
+    let result: JSX.Element[] = [];
+    if (obj) {
+      const belong = obj.belong;
+      const [off, offLst] = getBattleFactor(G, dualPlayerID(belong), true, CId);
+      const [def, defLst] = getBattleFactor(G, belong, false, CId);
+      const relDef = def - off;
+      //show the detailed info for selected cell, otherwise only cell in danger
+      if (selected) {
+        result = result.concat(
+          offLst
+            .map((id) => gTranslate(renderStr('⚔️', 0.4), CId2Pos(id).x - 0.3, CId2Pos(id).y - 0.3))
+            .concat(defLst.map((id) => gTranslate(renderStr('🛡️', 0.4), CId2Pos(id).x - 0.3, CId2Pos(id).y - 0.3))),
+        );
+      }
+      if (relDef < 0) {
+        result = result.concat(
+          gTranslate(renderStr(defState(relDef), 0.4), CId2Pos(CId).x + 0.3, CId2Pos(CId).y - 0.3),
+        );
+      }
+      return result;
+    }
+  }
+  const boardRef = useRef(null);
+  //map move ui
+  /*const [mapPos, setMapPos] = useState<Position>({ x: 0, y: 0 })
+  const [mapScale, setMapScale] = useState<number>(1)
+  //const [onDrag, setOnDrag]=useState<boolean>(false)
+  
+  
+   const gestureBind = useGesture(
+    {
+      onDrag: (state) => {
+        const e = state.event
+        const svg = e.target as SVGAElement
+        const CTM = svg.getScreenCTM()
+        if (CTM) {
+          const move = state.offset
+          const dx = move[0] / (CTM.a)
+          const dy = move[1] / (CTM.d)
+          setMapPos({ x: dx, y: dy });
+        }
+      },
+      onWheel: (state) => {
+        const evt =state.event;
+        evt.preventDefault();
+        const spd = 0.0007
+        
+        const newScale = mapScale * (1 - spd * state.movement[1])
+        //if (newScale > 0.5 && newScale < 10) {  }
+        setMapScale(newScale);
+      },
+      onPinch: (state) => {
+
+        const newScale = state.offset[0]
+        //if (newScale > 0.5 && newScale < 10) {  }
+
+        setMapScale(newScale);
+      },
+
+    },
+    {
+      target: boardRef,
+      eventOptions: { passive: false },
+      preventDefault: true,
+    }
+  ); */
```

Comment:
Please remove dead code.

Reply:
this code is for mobile phone UI control

Reply:
how to add [this libarary](https://github.com/pmndrs/use-gesture)? The simply installing doesn't work